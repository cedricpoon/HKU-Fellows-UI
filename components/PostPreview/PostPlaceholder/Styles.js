import nbColor from "hkufui/native-base-theme/variables/commonColor";
import { post } from 'hkufui/theme/grid'
import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  placeholder: {
    marginVertical: post.placeholderVerticalMargin
  },
  subHeadline: {
    flexDirection: 'row',
    marginTop: post.newLineMargin
  },
  button: {
    borderColor: nbColor.listNoteColor,
    alignSelf: 'center',
    marginTop: post.newLineMargin * 2,
    height: null
  }
});
