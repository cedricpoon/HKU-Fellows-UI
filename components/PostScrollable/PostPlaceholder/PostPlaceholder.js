import React from 'react';
import PropTypes from "prop-types";
import { Body as NBBody, Text, Icon, View, Button } from 'native-base';
import * as Animatable from 'react-native-animatable';

import styles from './Styles';
import peerStyles from '../PostPreview/Styles';
import { FADE_IN_DURATION } from '../PostScrollable';

const Body = Animatable.createAnimatableComponent(NBBody);

class PostPlaceholder extends React.Component {
  render() {
    const { headline, subHeadline, icon, headlineColor, button } = this.props;

    return (
      <Body style={styles.placeholder} animation="fadeIn" duration={FADE_IN_DURATION}>
        <Text style={{color: headlineColor}}>{headline}</Text>
        {subHeadline && (
          <View style={styles.subHeadline}>
            <Text note>
              {subHeadline}
            </Text>
            {icon && (
              <Icon
                name={icon.name}
                type={icon.type}
                active={icon.active}
                style={[peerStyles.badge, peerStyles.note]}
              />
            )}
          </View>
        )}
        {button && (
          <Button bordered style={styles.button} onPress={button.onPress}>
            <Text note>{button.text}</Text>
          </Button>
        )}
      </Body>
    );
  }
}

PostPlaceholder.propTypes = {
  headline: PropTypes.string.isRequired,
  headlineColor: PropTypes.string,
  subHeadline: PropTypes.string,
  icon: PropTypes.shape({
    name: PropTypes.string.isRequired,
    type: PropTypes.string,
    active: PropTypes.bool
  }),
  button: PropTypes.shape({
    text: PropTypes.string.isRequired,
    onPress: PropTypes.func
  })
}

export default PostPlaceholder;
