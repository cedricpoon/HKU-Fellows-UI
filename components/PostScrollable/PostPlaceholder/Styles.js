import { post } from "hkufui/theme/grid"
import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  placeholder: {
    marginVertical: post.placeholderVerticalMargin
  },
  subHeadline: {
    flexDirection: 'row',
    flex: 1,
    paddingTop: post.newLinePadding
  }
});
