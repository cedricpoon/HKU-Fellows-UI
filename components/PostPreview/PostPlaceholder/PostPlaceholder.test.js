import React from 'react';
import { shallow } from 'enzyme';

import PostPlaceholder from './PostPlaceholder';

describe('Testing PostPlaceholder Component', () => {
  it('renders as expected with required props', () => {
    const wrapper = shallow(
      <PostPlaceholder headline='' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with subHeadline', () => {
    const wrapper = shallow(
      <PostPlaceholder headline='' subHeadline='' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with headline colored', () => {
    const wrapper = shallow(
      <PostPlaceholder headline='' headlineColor='green' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with subHeadline and icon and button', () => {
    const wrapper = shallow(
      <PostPlaceholder headline='' subHeadline='' icon={{name:'apps'}} button={{text: ''}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
