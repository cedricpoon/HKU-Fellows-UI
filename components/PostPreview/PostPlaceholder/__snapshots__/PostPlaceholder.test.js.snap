// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostPlaceholder Component renders as expected with headline colored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostPlaceholder
    headline=""
    headlineColor="green"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": "fadeIn",
      "children": Array [
        <Styled(Text)
          style={
            Object {
              "color": "green",
            }
          }
        >
          
        </Styled(Text)>,
        undefined,
        undefined,
      ],
      "delay": 0,
      "direction": "normal",
      "duration": 250,
      "easing": undefined,
      "iterationCount": 1,
      "iterationDelay": 0,
      "onAnimationBegin": [Function],
      "onAnimationEnd": [Function],
      "onTransitionBegin": [Function],
      "onTransitionEnd": [Function],
      "style": Object {
        "marginVertical": 25,
      },
      "transition": undefined,
      "useNativeDriver": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "green",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": "fadeIn",
        "children": Array [
          <Styled(Text)
            style={
              Object {
                "color": "green",
              }
            }
          >
            
          </Styled(Text)>,
          undefined,
          undefined,
        ],
        "delay": 0,
        "direction": "normal",
        "duration": 250,
        "easing": undefined,
        "iterationCount": 1,
        "iterationDelay": 0,
        "onAnimationBegin": [Function],
        "onAnimationEnd": [Function],
        "onTransitionBegin": [Function],
        "onTransitionEnd": [Function],
        "style": Object {
          "marginVertical": 25,
        },
        "transition": undefined,
        "useNativeDriver": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "green",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing PostPlaceholder Component renders as expected with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostPlaceholder
    headline=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": "fadeIn",
      "children": Array [
        <Styled(Text)
          style={
            Object {
              "color": undefined,
            }
          }
        >
          
        </Styled(Text)>,
        undefined,
        undefined,
      ],
      "delay": 0,
      "direction": "normal",
      "duration": 250,
      "easing": undefined,
      "iterationCount": 1,
      "iterationDelay": 0,
      "onAnimationBegin": [Function],
      "onAnimationEnd": [Function],
      "onTransitionBegin": [Function],
      "onTransitionEnd": [Function],
      "style": Object {
        "marginVertical": 25,
      },
      "transition": undefined,
      "useNativeDriver": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "style": Object {
            "color": undefined,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": "fadeIn",
        "children": Array [
          <Styled(Text)
            style={
              Object {
                "color": undefined,
              }
            }
          >
            
          </Styled(Text)>,
          undefined,
          undefined,
        ],
        "delay": 0,
        "direction": "normal",
        "duration": 250,
        "easing": undefined,
        "iterationCount": 1,
        "iterationDelay": 0,
        "onAnimationBegin": [Function],
        "onAnimationEnd": [Function],
        "onTransitionBegin": [Function],
        "onTransitionEnd": [Function],
        "style": Object {
          "marginVertical": 25,
        },
        "transition": undefined,
        "useNativeDriver": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "style": Object {
              "color": undefined,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing PostPlaceholder Component renders as expected with subHeadline 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostPlaceholder
    headline=""
    subHeadline=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": "fadeIn",
      "children": Array [
        <Styled(Text)
          style={
            Object {
              "color": undefined,
            }
          }
        >
          
        </Styled(Text)>,
        "",
        undefined,
      ],
      "delay": 0,
      "direction": "normal",
      "duration": 250,
      "easing": undefined,
      "iterationCount": 1,
      "iterationDelay": 0,
      "onAnimationBegin": [Function],
      "onAnimationEnd": [Function],
      "onTransitionBegin": [Function],
      "onTransitionEnd": [Function],
      "style": Object {
        "marginVertical": 25,
      },
      "transition": undefined,
      "useNativeDriver": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "style": Object {
            "color": undefined,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "",
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": "fadeIn",
        "children": Array [
          <Styled(Text)
            style={
              Object {
                "color": undefined,
              }
            }
          >
            
          </Styled(Text)>,
          "",
          undefined,
        ],
        "delay": 0,
        "direction": "normal",
        "duration": 250,
        "easing": undefined,
        "iterationCount": 1,
        "iterationDelay": 0,
        "onAnimationBegin": [Function],
        "onAnimationEnd": [Function],
        "onTransitionBegin": [Function],
        "onTransitionEnd": [Function],
        "style": Object {
          "marginVertical": 25,
        },
        "transition": undefined,
        "useNativeDriver": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "style": Object {
              "color": undefined,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "",
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing PostPlaceholder Component renders as expected with subHeadline and icon and button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostPlaceholder
    button={
      Object {
        "text": "",
      }
    }
    headline=""
    icon={
      Object {
        "name": "apps",
      }
    }
    subHeadline=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": "fadeIn",
      "children": Array [
        <Styled(Text)
          style={
            Object {
              "color": undefined,
            }
          }
        >
          
        </Styled(Text)>,
        "",
        <Styled(Button)
          bordered={true}
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "#808080",
              "height": null,
              "marginTop": 20,
            }
          }
        >
          <Styled(Text)
            note={true}
          >
            
          </Styled(Text)>
        </Styled(Button)>,
      ],
      "delay": 0,
      "direction": "normal",
      "duration": 250,
      "easing": undefined,
      "iterationCount": 1,
      "iterationDelay": 0,
      "onAnimationBegin": [Function],
      "onAnimationEnd": [Function],
      "onTransitionBegin": [Function],
      "onTransitionEnd": [Function],
      "style": Object {
        "marginVertical": 25,
      },
      "transition": undefined,
      "useNativeDriver": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "style": Object {
            "color": undefined,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": true,
          "children": <Styled(Text)
            note={true}
          >
            
          </Styled(Text)>,
          "onPress": undefined,
          "style": Object {
            "alignSelf": "center",
            "borderColor": "#808080",
            "height": null,
            "marginTop": 20,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "note": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": "fadeIn",
        "children": Array [
          <Styled(Text)
            style={
              Object {
                "color": undefined,
              }
            }
          >
            
          </Styled(Text)>,
          "",
          <Styled(Button)
            bordered={true}
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#808080",
                "height": null,
                "marginTop": 20,
              }
            }
          >
            <Styled(Text)
              note={true}
            >
              
            </Styled(Text)>
          </Styled(Button)>,
        ],
        "delay": 0,
        "direction": "normal",
        "duration": 250,
        "easing": undefined,
        "iterationCount": 1,
        "iterationDelay": 0,
        "onAnimationBegin": [Function],
        "onAnimationEnd": [Function],
        "onTransitionBegin": [Function],
        "onTransitionEnd": [Function],
        "style": Object {
          "marginVertical": 25,
        },
        "transition": undefined,
        "useNativeDriver": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "style": Object {
              "color": undefined,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": true,
            "children": <Styled(Text)
              note={true}
            >
              
            </Styled(Text)>,
            "onPress": undefined,
            "style": Object {
              "alignSelf": "center",
              "borderColor": "#808080",
              "height": null,
              "marginTop": 20,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "note": true,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
